python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -1,30 +1,173 @@
+    except ValueError:
+        raise DagsterInvariantViolationError(
+            'Could not parse execution_date "{execution_date_str}". Please use datetime'
+            " format compatible with  dateutil.parser.parse.".format(
+                execution_date_str=execution_date_str,
+            )
+        )
+    except OverflowError:
+        raise DagsterInvariantViolationError(
+            'Date "{execution_date_str}" exceeds the largest valid C integer on the system.'.format(
+                execution_date_str=execution_date_str,
+            )
+        )
+    return execution_date
+
+
+def _parse_execution_date_for_asset(
+    dag: DAG, run_tags: Mapping[str, str]
+) -> Optional[datetime.datetime]:
+    execution_date_str = run_tags.get("dagster/partition")
+    if not execution_date_str:
+        raise DagsterInvariantViolationError("dagster/partition is not set")
+    execution_date = parse(execution_date_str)
+    execution_date = execution_date.replace(tzinfo=pytz.timezone(dag.timezone.name))
+    return execution_date
+
+
+def airflow_db(context: InitResourceContext) -> None:
+    airflow_home_path = os.path.join(tempfile.gettempdir(), f"dagster_airflow_{context.run_id}")
+    os.environ["AIRFLOW_HOME"] = airflow_home_path
+    os.makedirs(airflow_home_path, exist_ok=True)
+    with Locker(airflow_home_path):
+        airflow_initialized = os.path.exists(f"{airflow_home_path}/airflow.db")
+        # because AIRFLOW_HOME has been overriden airflow needs to be reloaded
+        if is_airflow_2_loaded_in_environment():
+            importlib.reload(airflow.configuration)
+            importlib.reload(airflow.settings)
+            importlib.reload(airflow)
+        else:
+            importlib.reload(airflow)
+        if not airflow_initialized:
+            db.initdb()
+            create_airflow_connections(
+                [Connection(**c) for c in context.resource_config["connections"]]
+            )
+
+
+def _create_airflow_db_resource(connections: Optional[List[Connection]] = None):
+    serialized_connections = serialize_connections(connections)
+    airflow_db_resource_def = ResourceDefinition(
+        resource_fn=airflow_db,
+        required_resource_keys={"dag"},
+        config_schema={
+            "connections": Field(
+                Array(inner_type=dict),
+                default_value=serialized_connections,
+                is_required=False,
+            ),
+        },
+        description="Airflow DB used by wrapped airflow tasks",
+    )
+    return airflow_db_resource_def
+
+
+def get_dagrun(context: InitResourceContext) -> DagRun:
+    airflow_home_path = os.path.join(tempfile.gettempdir(), f"dagster_airflow_{context.run_id}")
+    os.makedirs(airflow_home_path, exist_ok=True)
+    with Locker(airflow_home_path):
+        dag = context.resources.dag
+        run_tags = context.dagster_run.tags if context.dagster_run else {}
+        if AIRFLOW_EXECUTION_DATE_STR in run_tags:
```

Comment:
maybe set execution_date to None above here. Code like this where variables are not explicitly declared in the outer scope always gives me the willies